Content.Server/Research/Systems/RoboticsConsoleSystem.cs

```diff
@@ -0,0 +1,157 @@
+using Content.Server.Administration.Logs;
+using Content.Server.DeviceNetwork;
+using Content.Server.DeviceNetwork.Systems;
+using Content.Server.Radio.EntitySystems;
+using Content.Server.Research.Components;
+using Content.Shared.Access.Systems;
+using Content.Shared.Database;
+using Content.Shared.Research;
+using Robust.Server.GameObjects;
+using Robust.Shared.Timing;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Content.Server.Research.Systems;
+
+/// <summary>
+/// Handles UI and state receiving for the robotics control console.
+/// <c>BorgTransponderComponent<c/> broadcasts state from the station's borgs to consoles.
+/// </summary>
+public sealed class RoboticsConsoleSystem : EntitySystem
+{
+    [Dependency] private readonly AccessReaderSystem _access = default!;
+    [Dependency] private readonly DeviceNetworkSystem _deviceNetwork = default!;
+    [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly RadioSystem _radio = default!;
+    [Dependency] private readonly UserInterfaceSystem _ui = default!;
+
+    // almost never timing out more than 1 per tick so initialize with that capacity
+    private List<string> _removing = new(1);
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RoboticsConsoleComponent, EntityUnpausedEvent>(OnUnpaused);
+        SubscribeLocalEvent<RoboticsConsoleComponent, DeviceNetworkPacketEvent>(OnPacketReceived);
+        Subs.BuiEvents<RoboticsConsoleComponent>(RoboticsConsoleUiKey.Key, subs =>
+        {
+            subs.Event<BoundUIOpenedEvent>(OnOpened);
+            subs.Event<RoboticsConsoleDisableMessage>(OnDisable);
+            subs.Event<RoboticsConsoleDestroyMessage>(OnDestroy);
+            // TODO: camera stuff
+        });
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var now = _timing.CurTime;
+        foreach (var comp in EntityQuery<RoboticsConsoleComponent>())
```

Comment:
use EntityQueryEnumerator

Reply:
uid isnt used tho

Reply:
Only 1 of the overloads even has it + just dump it noob. IEnumerable slow.

Reply:
ok microsoft java moment